### PR TITLE
Decline: add native support

### PR DESCRIPTION
**Pull request by kubukoz**
I think this is the last module we can actually add Native support to - all the others have some JVM-only dependencies we won't be able to get rid of.

### CODE REVIEW

**Comment by kubukoz** [comment]
can't reproduce the failure locally :(

File: build.sbt [hunk 1]
```diff
@@ -781,7 +782,7 @@ lazy val Dependencies = new {
   }
 
   object Decline {
-    val core = Def.setting("com.monovore" %%% "decline" % "2.3.0")
+    val core = Def.setting("com.monovore" %%% "decline" % "2.3.1")
     val effect = Def.setting("com.monovore" %%% "decline-effect" % "2.3.0")
```
Comment (daddykotex) [inline build.sbt:786]: should we bump the one below also? or introduce a val to share

**Comment by plokhotnyuk** [comment]
> can't reproduce the failure locally :(

Please consider replacing the following regex by some simple validation function:
https://github.com/disneystreaming/smithy4s/blob/main/modules/core/test/src-jvm/smithy4s/TimestampSpec.scala#L124

File: build.sbt (hunk 1 above)
Comment (kubukoz) [reply build.sbt:786]: good catch, it is indeed used in docs. I'll bring in a val

File: build.sbt [hunk 2]
```diff
@@ -47,27 +47,27 @@ lazy val root = project
...
+  protocol,
+  protocolTests,
+  openapi,
+  `aws-kernel`,
+  aws,
+  `aws-http4s`,
+  `codegen-cli`,
+  dynamic,
+  testUtils
+).flatMap(_.projectRefs)
```
Comment (lewisjkl) [inline build.sbt:70]: This is some 3022 thinking right here 🎉

**Review by lewisjkl — APPROVED** [review]

**Pull request merged**